woodwork/serializers/serializer_base.py

```diff
@@ -0,0 +1,203 @@
+import datetime
+import json
+import os
+import tarfile
+import tempfile
+
+from woodwork.accessor_utils import _is_dask_dataframe, _is_koalas_dataframe
+from woodwork.logical_types import LatLong
+from woodwork.s3_utils import get_transport_params, use_smartopen
+from woodwork.type_sys.utils import _get_ltype_class, _get_specified_ltype_params
+from woodwork.utils import _is_s3, _is_url
+
+SCHEMA_VERSION = "11.3.0"
```

Comment:
Do we want to increase the version as a result of adding custom names for the data subdirectory and typing info?

Reply:
Good question. I debated about that a little myself. Ultimately, I landed on leaving it the same since we aren't changing anything in the JSON data use to store the typing information. Also, older versions of WW will be able to deserialize data serialized with this version when the default values for filename, data subdirectory and typing info file are used.

Not exactly sure what's best here though.

Reply:
Yeah, I'm not entirely sure either. There's also the case of saving the data and typing info using custom names with this version and not being able to load using previous versions.

Reply:
Yeah, we would lose the ability to deserialize with older versions in cases where the defaults are changed.

@rwedge @tamargrey Any thoughts on what is appropriate here? Should we keep the SCHEMA_VERSION the same or update with the changes in this PR?

Reply:
The goal here is that this wouldn't be a breaking change, right? If previously serialized dataframes (with schema 11.3.0) can still be deserialized now, and older versions of Woodwork can deserialize from this version of Woodwork with the default file names, that feels like we should keep the schema the same.

Reply:
Yeah, the idea here was that the same serialize and deserialize API calls that worked before would still work after this update and achieve the same result. I think that was accomplished at the level of `df.ww.to_disk` and `read_woodwork_table`. There were some breaking API changes in terms of the other utility functions used in the process that have now been removed from the public API.

The main wrinkle is that this PR implements additional capability that is not backward compatible.